pkg/file/redundancy/getter/strategies.go

```diff
@@ -30,15 +30,39 @@ const (
 )
 
 // GetParamsFromContext extracts the strategy and strict mode from the context
-func GetParamsFromContext(ctx context.Context) (s Strategy, strict bool, fetcherTimeout time.Duration) {
-	s, _ = ctx.Value(strategyKey{}).(Strategy)
-	strict, _ = ctx.Value(modeKey{}).(bool)
-	fetcherTimeout, _ = ctx.Value(fetcherTimeoutKey{}).(time.Duration)
-	return s, strict, fetcherTimeout
+func GetParamsFromContext(ctx context.Context) (s Strategy, strict bool, fetcherTimeout time.Duration, err error) {
+	var ok bool
+	s, strict, fetcherTimeoutVal := NONE, true, "30s"
```

Comment:
these defaults should be defined in the api package.
After parsing the headers, and before calling `SetParamsInContext`, the defaults should be assigned if the values were not passed in the header.

Reply:
even though it is not the API's defaults but generic defaults for the functionality?

Reply:
the pkg can export the defaults and whichever pkg is setting the context params, they can use those exported defaults if not values are supplied.